### PR TITLE
python2Packages.cryptography: include missing patch file

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
During https://github.com/NixOS/nixpkgs/pull/155185 a `patch` file which should have been moved with `pythonPackages.cryptography` was missed. Even with the missing patch file included the package still didn't build due to dependencies `iso8601` and `pytest-forked` not building against `python2`, so I dropped back to the latest supported versions for those packages. I'm able to maintain this if desired.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by zombiezen — APPROVED** [review]
Just hit this myself, would love to get this merged in.

**Comment by veprbl** [comment]
Change looks good to me.

Courtesy pings to @fabaff and @dotlambda in case they don't want to be in `meta.maintainers` for the python 2 package.

File: pkgs/development/python2-modules/pytest-forked/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    py
+  ];
+
+  checkInputs = [ pytestCheckHook ];
+
+  meta = {
+    description = "Run tests in isolated forked subprocesses";
+    homepage = "https://github.com/pytest-dev/pytest-forked";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ dotlambda ];
```
Comment (dotlambda) [inline pkgs/development/python2-modules/pytest-forked/default.nix:35]:
```suggestion
    maintainers = with lib.maintainers; [ aanderse ];
```

**Review by dotlambda — COMMENTED** [review]
For which leaf package is this needed?
We're trying to get rid of Python 2: https://github.com/NixOS/nixpkgs/issues/148779

**Comment by FRidh** [comment]
Pretty sure they don't want to maintain the Python 2 packages. Whoever needs these should really add themselves as maintainer.

**Comment by aanderse** [comment]
I'm free of `python` 2!

**Pull request closed without merging**